apps/web/pages/auth/login.tsx

```diff
@@ -59,22 +67,23 @@ export default function Login({
     [ErrorCode.IncorrectTwoFactorCode]: `${t("incorrect_2fa_code")} ${t("please_try_again")}`,
     [ErrorCode.InternalServerError]: `${t("something_went_wrong")} ${t("please_try_again_and_contact_us")}`,
     [ErrorCode.ThirdPartyIdentityProviderEnabled]: t("account_created_with_identity_provider"),
+    [ErrorCode.InvalidCommunityLogin]: `${t("something_went_wrong")} ${t("signing_in_community")}`,
   };
 
   const telemetry = useTelemetry();
 
-  let callbackUrl = typeof router.query?.callbackUrl === "string" ? router.query.callbackUrl : "";
+  const queryParams = querySchema.parse(router.query);
 
-  if (/"\//.test(callbackUrl)) callbackUrl = callbackUrl.substring(1);
+  if (/"\//.test(queryParams.callbackUrl)) queryParams.callbackUrl = queryParams.callbackUrl.substring(1);
 
   // If not absolute URL, make it absolute
-  if (!/^https?:\/\//.test(callbackUrl)) {
-    callbackUrl = `${WEBAPP_URL}/${callbackUrl}`;
+  if (!/^https?:\/\//.test(queryParams.callbackUrl)) {
+    queryParams.callbackUrl = `${WEBAPP_URL}/${queryParams.callbackUrl}`;
   }
 
-  const safeCallbackUrl = getSafeRedirectUrl(callbackUrl);
+  const safeCallbackUrl = getSafeRedirectUrl(queryParams.callbackUrl);
 
-  callbackUrl = safeCallbackUrl || "";
+  queryParams.callbackUrl = safeCallbackUrl || "";
```

Comment:
Above lines is just moving from simple query checking to Zod schema, nothing changed really in terms of logic.